File: Frequency Game - GFG/frequency-game.py
#User function Template for python3

from collections import Counter
def LargButMinFreq(arr,n):
    #code here
    c=Counter(arr)
    x=min(c.values())
    ans=0
    for i in c:
        if c[i]==x:
            ans=max(i,ans)
    return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

t = int(input())
#Iterating over test cases
for _ in range(t):
    n = int(input())
    arr = [int(x) for x in input().split()]
    print(LargButMinFreq(arr, n))
# } Driver Code Ends

File: Frequency Game - GFG/frequency-game.java
//{ Driver Code Starts
// Initial Template for Java

import java.util.*;
import java.io.*;

// Position this line where user code will be pasted.

// Driver class with main function
class GFG {
    // Main function
    public static void main(String[] args)throws IOException {
        BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(in.readLine());

        // Iterating over testcases
        while (t-- > 0) {
            int n = Integer.parseInt(in.readLine());
            int arr[] = new int[n];

            String s[]=in.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) arr[i] = Integer.parseInt(s[i]);

            Solution obj = new Solution();
            System.out.println(obj.LargButMinFreq(arr, n));
        }
    }
}
// } Driver Code Ends


// User function Template for Java

// Helper class to find largest number with minimum frequency
class Solution {
    // Function to find largest number with minimum frequency
    public static int LargButMinFreq(int arr[], int n) {
        // Your code here
        HashMap<Integer, Integer> counter = new HashMap<>();
        for (int i = 0; i < n; i++) {
            counter.put(arr[i], counter.getOrDefault(arr[i], 0) + 1);
        }
        
        int minFreq = Integer.MAX_VALUE;
        for (int freq : counter.values()) {
            minFreq = Math.min(minFreq, freq);
        }
        
        int answer = 0;
        for (int num : counter.keySet()) {
            if (counter.get(num) == minFreq) {
                answer = Math.max(answer, num);
            }
        }
        
        return answer;
    }
}